app/controllers/shopify_app/callback_controller.rb

```diff
@@ -89,6 +89,14 @@ def shop_name
       auth_hash.uid
     end
 
+    def offline_access_token
+      ShopifyApp::SessionRepository.retrieve_shop_session_by_shopify_domain(shop_name)&.token
+    end
```

Comment:
Maybe we should stick with "shop" and "user" rather than "offline" and "online" since that's what we're using elsewhere in the code.

Reply:
That's a good point. I thought this approach would help us slowly move towards adopting the offline and online terminology.

Reply:
If I can interject here. You guys hit the nail on the head. To me, some outside plebe with no idea what is going on, this whole concept of Shop versus User has never really been clearly explained anywhere, by anyone. The explanations as they are, remain weak and abstract. How about some case studies. You never explain the difference between a User and a User from the sense of Shopify Plus either. So that is confusing. And now to go, online, offline in lingo, say what? 

We in the community suffer through so many bike shed paintings already. Avoiding those at all costs is doing a huge favor! It used to me myshopify_domain, then it was shopify_domain; it is SHOPIFY_API_SECRET but someone in JS world wants it to be SHOPIFY_API_SECRET_KEY, and I am like OH FOR THE LOVE OF PAINTING YAKS AND SHAVING THEIR SHEDS.

Just my two cents. Have at 'er.

Reply:
Now that we've chatted about it, I get why it'd be nice to call these "offline_access_token" and "online_access_token". Most of our use of "shop" and "user" is separate from this.

Having said that, we do have some follow-up work, for example, with the use of "shop_token" in the webhooks manager code.